File: src/main/java/sawyern/cookiebot/controllers/AccountController.java
package sawyern.cookiebot.controllers;

import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import sawyern.cookiebot.models.dto.AccountDto;
import sawyern.cookiebot.exception.CookieException;
import sawyern.cookiebot.services.AccountService;

import javax.transaction.Transactional;

@RestController
@RequestMapping(value = "/accounts")
@RequiredArgsConstructor
public class AccountController implements GenericController {

    private final AccountService accountService;

    @PutMapping
    @Transactional
    public ResponseEntity<String> registerNewUser(
            @RequestBody AccountDto accountDto
            ) {
        try {
            accountService.registerAccount(accountDto);
            return new ResponseEntity<>(HttpStatus.OK);
        } catch (CookieException e) {
            return new ResponseEntity<>(e.getMessage(), e.getHttpStatus());
        }
    }

    @GetMapping
    public ResponseEntity<String> test() {
        return new ResponseEntity<>("hi", HttpStatus.OK);
    }
}

File: src/main/kotlin/sawyern/cookiebot/controllers/AccountController.kt
package sawyern.cookiebot.controllers;

import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping;
import sawyern.cookiebot.models.Account
import sawyern.cookiebot.services.AccountService

@Controller
@RequestMapping("/api/cookiebot/v1/accounts")
class AccountController (
        private val accountService: AccountService
) {
    @GetMapping
    fun getAllAccounts(): ResponseEntity<List<Account>> {
        return ResponseEntity.ok(accountService.findAll())
    }
}